### PR TITLE
feat: apply name proposal to DictionaryGetAll

### PR DESCRIPTION
**Pull request by malandis**
In line with the name proposal, this renames DictionaryGetAll* to
DictionaryFetch*.

### CODE REVIEW

File: proto/cacheclient.proto [hunk 1]
```diff
@@ -70,7 +70,7 @@ message _DictionaryGetResponse {
   string message = 2;
 }
 
-message _DictionaryGetAllRequest {
+message _DictionaryFetchRequest {
```
Comment (anp13) [inline proto/cacheclient.proto:73]: We also need to update the `DictionaryGetResponse` and `DictionaryFetchResponse` to the `Found/Missing` format like we did for `SetFetch` and also remove the `ECacheResult` from all the dictionary api responses.

File: proto/cacheclient.proto [hunk 2]
```diff
@@ -58,53 +58,66 @@ message _SetResponse {
...
+  message _Found {
+    repeated _DictionaryGetResponsePart items = 1;
+  }
+
+  message _Missing {}
+
+  oneof dictionary {
+    _Found found = 1;
+    _Missing missing = 2;
+  }
```
Comment (malandis) [inline proto/cacheclient.proto:79]: I applied the `_Found` (`_Missing`) pattern to the dictionary overall but not the unary items within.

File: proto/cacheclient.proto (hunk 1 above)
Comment (malandis) [reply proto/cacheclient.proto:82]: Added `Found/Missing` in caf8c1a.

File: proto/cacheclient.proto [hunk 3]
```diff
@@ -58,53 +58,66 @@ message _SetResponse {
...
 message _DictionaryDeleteRequest {
   message Some {
-    repeated bytes keys = 1;
+    repeated bytes fields = 1;
   }
   message All { }
 
   bytes dictionary_name = 1;
-  oneof delete_keys {
+  oneof delete_fields {
```
Comment (anp13) [inline proto/cacheclient.proto:120]: nit: we can call this `delete` to be consistent with the set and list apis. We use `difference` in SetDifferenceRequest and for list apis also I have used `erase` in ListEraseRequest and `remove` in ListRemoveRequest. 

https://github.com/momentohq/client_protos/pull/71/files

**Review by anp13 — COMMENTED** [review]
Looks good just one comment. 

Do you think we should change `Found` to `Ok` especially for update api responses? `Found` seems weird in api responses which do not return anything.

File: proto/cacheclient.proto (hunk 3 above)
Comment (malandis) [reply proto/cacheclient.proto]: Fixed in a332286

**Comment by malandis** [comment]
> Looks good just one comment.
> 
> Do you think we should change `Found` to `Ok` especially for update api responses? `Found` seems weird in api responses which do not return anything.

Do you mean for the `Set` APIs?

**Comment by anp13** [comment]
> > Looks good just one comment.
> > Do you think we should change `Found` to `Ok` especially for update api responses? `Found` seems weird in api responses which do not return anything.
> 
> Do you mean for the `Set` APIs?

Yes. It only applies to the `SetDifference` API I think. I think it is ok to keep it as `Found` for now since we are not using it in the SDK yet.

**Review by anp13 — APPROVED** [review]

**Pull request merged**